File: BinarySearchAlgo/Binarysearch.c
#include<stdio.h>
int main(){
    int n;
    scanf("%d",&n);
    int arr[n];
    for(int i=0;i<n;i++){
        scanf("%d",&arr[i]);
    }
    int low=0,high=n-1,s;
    scanf("%d",&s);
    while(low<=high){
        int mid = (low+high)/2;

        if(arr[mid]==s){
            printf("%d Found at Index %d",s,mid);
            return 0;
        }
        else if(s>arr[mid]){
            low=mid+1;
        }
        else{
            high=mid-1;
        }
    }
    printf("Not Found");
    return 0;
    
}

File: BinarySearchAlgo/Binarysearch.java
import java.util.Scanner;

public class Binarysearch {
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int arr[] = new int[n];

        for(int i=0;i<n;i++){
            arr[i] = sc.nextInt();
        }
        int low=0,high=n-1,s;
        s=sc.nextInt();
        
        while(low<=high){
            int mid = (low+high)/2;

            if(arr[mid]==s){
                System.out.println(s+" Found at Index "+mid);
                return;
            }
            else if(s>arr[mid]){
                low=mid+1;
            }
            else{
                high=mid-1;
            }
        }
        System.out.println("Not Found");
    }
}